File: GanttView/RadGanttViewTruckSchedulerExample/RadGanttViewExample/CustomGanttViewTaskItemElement.cs
using System;
using System.Data;
using Telerik.WinControls.UI;

namespace RadGanttViewExample
{
    public class CustomGanttViewTaskItemElement : GanttViewTaskItemElement
    {
        public CustomGanttViewTaskItemElement(GanttViewGraphicalViewElement owner)
            : base(owner)
        { }

        protected override GanttGraphicalViewBaseTaskElement CreateTaskElement()
        {
            return new CustomGanttGraphicalViewBaseTaskElement();
        }

        protected override Type ThemeEffectiveType
        {
            get { return typeof(GanttViewTaskItemElement); }
        }
    }
}

File: GanttView/RadGanttViewTruckSchedulerExample/RadGanttViewExampleVB/CustomGanttViewTaskItemElement.vb
Imports Telerik.WinControls.UI

Public Class CustomGanttViewTaskItemElement
    Inherits GanttViewTaskItemElement

    Public Sub New(ByVal owner As GanttViewGraphicalViewElement)
        MyBase.New(owner)
    End Sub

    Protected Overrides Function CreateTaskElement() As GanttGraphicalViewBaseTaskElement
        Return New CustomGanttGraphicalViewBaseTaskElement()
    End Function

    Protected Overrides ReadOnly Property ThemeEffectiveType As Type
        Get
            Return GetType(GanttViewTaskItemElement)
        End Get
    End Property
End Class
